hooks/shapediver/viewer/interaction/utils/patternUtils.ts

```diff
@@ -0,0 +1,137 @@
+import { IOutputApi, ITreeNode, OutputApiData } from "@shapediver/viewer";
+
+/**
+ * The type for the name filter pattern.
+ * 
+ * This is an object where the key is the output ID.
+ * The value is an array of the patterns to match the node names (which is an array of strings).
+ * 
+ * The matching is done by checking if the node name matches the pattern. This can be on any level of the node hierarchy.
+ * 
+ * Example:
+ * ```json
+ * {
+ *    "OUTPUT_ID": [
+ * 	 	["Node_a", "Node_b"], // matches nodes with the name "Node_a.Node_b" in the output with the ID "OUTPUT_ID"
+ * 	 	["Node_a", "Node_c"], // matches nodes with the name "Node_a.Node_c" in the output with the ID "OUTPUT_ID"
+ * 	 	["Node_a", "Node_d", "Node_e"] // matches nodes with the name "Node_a.Node_d.Node_e" in the output with the ID "OUTPUT_ID"
+ *   ]
+ * }
+ * 
+ */
+export type NameFilterPattern = { [key: string]: string[][] };
+
+/**
+ * Process the nameFilter and update the pattern.
+ * 
+ * The name filter is an array of strings where each string is a pattern to match the node names.
+ * The first part of the pattern is the output name.
+ * The rest of the pattern are the node names separated by a dot.
+ * The node names can contain "*" as a wildcard to match any node name or any part of the node name.
+ * 
+ * @param nameFilter The name filter to process.
+ * @param outputIdsToNamesMapping The mapping of the output IDs to the output names.
+ * @returns The pattern object with updated patterns.
+ */
+export const processPattern = (nameFilter: string[], outputIdsToNamesMapping: { [key: string]: string }): NameFilterPattern => {
+	const pattern: {
+		[key: string]: string[][];
+	} = {};
+
+	// we iterate over the name filter array
+	// we store the result with the output ID as the key and an array of patterns as the value
+	for (let i = 0; i < nameFilter.length; i++) {
+		const parts = nameFilter[i].split(".");
+		const outputName = parts[0];
+		
+		// find the output Ids that match the output name
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		const outputIds = Object.entries(outputIdsToNamesMapping).filter(([id, name]) => name === outputName).map(([outputId, _]) => outputId);
```

Comment:
Small optimization possibility: Outside of the loop we could create an object holding the inverse mapping of `outputIdsToNamesMapping`

Reply:
I just realized that this doesn't support wildcards for the output name (the first component of the name filter). Is this a bug or a feature? @MajorMeerkatThe3rd @mathieu-shapediver

Reply:
@snabela I was not aware that this was even an option, so from my side, the implementation is correct.

Reply:
@mathieu-shapediver let's make sure to document it like this. We might add additional validation for the name filters in Grasshopper (no wildcards allowed in the first component).